fix #6770 CRLF for cmd.exe

kalefranz opened the pull request:
fix #6770

mingwandroid commented:
If you open the file in text-mode Windows will take care of this for you. I am unaware of Windows having any trouble running batch files that use LF though. I have never seen that to be the case.

kalefranz commented:
I think the issue here was that the default mode for `NamedTemporaryFile` was binary.  I added in the mode explicitly in this PR, but what I added in is just the default.  At this point, since we're wrapping what gets written in `ensure_binary()`, I'm more comfortable making the value of `command_join` be `\r\n` than I am changing the mode of `NamedTemporaryFile` to text.
